src/firebolt/async_db/cursor.py

```diff
@@ -267,6 +272,13 @@ def _reset(self) -> None:
         self._idx = 0
         self._row_sets = []
         self._next_set_idx = 0
+        self._query_id = ""
+
+    def _query_id_from_response_async(self, response: Response) -> str:
+        if response.headers.get("content-length", "") == "0":
+            raise InternalError("No response to asynchronous query.")
```

Comment:
nit: InternalError -> OperationalError

Reply:
Done.

Reply:
This test is now included. Thanks Stepan!